src/hooks/useIdp/index.js

```diff
@@ -0,0 +1,50 @@
+/**
```

Comment:
I'm not sure this is so generic it belongs here instead of under the login page. I think it could also use a rename to make it clear - `useIdpFromQuery` - because otherwise it seems as though it might provide additional information.